### DISCUSSION TITLE
How to modify the response when there are multiple plug-ins

### DISCUSSION THREAD

**Discussion by 1798972** [Kong/kong#8556] [Help] [answered]
**I have two plugins to handle the response, A first and B second.**<br/>
**This is how I usually handle it in plugin A.**<br/>
![image](https://user-images.githubusercontent.com/49218070/159218238-89a68c4a-f56e-4204-94a7-5d2fb8ab4cd4.png)
**When I finished processing in plug-in A, plug-in B still received the original response.**<br/>
![image](https://user-images.githubusercontent.com/49218070/159217991-fe76a798-874e-4226-a5dd-dbaebe8bb8e2.png)
![image](https://user-images.githubusercontent.com/49218070/159218046-2a2f602c-8d13-4d98-b4ca-1f4b9c139bef.png)
**When processing in plug-in B, I always receive the original data of plug-in A.**<br/>
![image](https://user-images.githubusercontent.com/49218070/159218188-ab355c50-b9cf-48e2-a4f0-8e91955b6afc.png)
---
Thanks

**Comment by locao** [comment, marked as answer]
Hi @1798972!

Kong PDK has functions for getting and setting the response body, you don't have to deal with chunks, end of body and whatnots. Please take a look at [kong.response.get_raw_body()](https://docs.konghq.com/gateway/2.8.x/pdk/kong.response/#kongresponseget_raw_body) and [kong.response.set_raw_body()](https://docs.konghq.com/gateway/2.8.x/pdk/kong.response/#kongresponseset_raw_bodybody).
